### PR TITLE
widget-box & list-layout components

### PR DESCRIPTION
**Pull request by MikiAsh**
This is the groundwork and basic setup for the component architecture.
More work is needed but let's establish the basics first

### CODE REVIEW

File: projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
+@import '../../style/common-imports';
+
+:host {
+  display: flex;
+  flex-direction: column;
+  background: #FFFFFF;
+  box-shadow: 0px 2px 1px rgba(0, 0, 0, 0.05), 0px 0px 1px rgba(0, 0, 0, 0.25);
+  border-radius: 8px;
+  margin: 24px; //DELETE ME
```
Comment (MikiAsh) [inline projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss:9]: Yes I know. this css is not finalised

File: projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+header {
+  height: 20px;
+  border: 1px solid lightgrey; //DELETE ME
+  margin: 16px 16px 0 16px;
+
+  b-subheading {
+    flex: 1;
+    line-height: 19.2px;
+  }
```
Comment (galloween) [inline projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss:21]: using line-height to align text vertically is a bad idea (if text will need to wrap, you will have a problem).

instead, add `flx-row-align-y` class to container (header).
and class `flx-grow` to subheading

File: projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  .header-right {
+    margin-right: 20px;
+    color: $grey-600;
+    line-height: 19.2px;
+  }
+}
+
+.content {
+  height: 142px;
```
Comment (galloween) [inline projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss:31]: // delete me too

File: projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  border: 1px solid lightgrey; //DELETE ME
+}
+
+b-button ::ng-deep button {
+  color: $grey-700;
+  font-size: 12px !important;
+  font-weight: 700 !important;
+  height: 19px !important;
+  line-height: 19.2px !important;
+}
```
Comment (galloween) [inline projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss:41]: `b-text-button` https://hibobio.github.io/bob-style/?path=/story/buttons--text-button
+ dont change height or line-height

**Review by galloween — COMMENTED** [review]
+ create a story so you can test it

File: projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  box-shadow: 0px 2px 1px rgba(0, 0, 0, 0.05), 0px 0px 1px rgba(0, 0, 0, 0.25);
+  border-radius: 8px;
+  margin: 24px; //DELETE ME
+  user-select: none;
+}
+
+header {
+  height: 20px;
+  border: 1px solid lightgrey; //DELETE ME
+  margin: 16px 16px 0 16px;
```
[the remark this answers is not in the document]
Comment (MikiAsh) [reply projects/ui-framework/src/lib/layout/widget-box/widget-box.component.scss:16]: "margin should be padding", why?

**Pull request merged**